File: CPP/valid_parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
     
         stack<char> stk;
   char x;
   for (int i=0; i<s.length(); i++) {
      if (s[i]=='('||s[i]=='['||s[i]=='{') {
         stk.push(s[i]);
         continue;
      }
      if (stk.empty())
         return false;
      switch (s[i]) {
      case ')':
         x = stk.top();
         stk.pop();
         if (x=='{' || x=='[')
            return false;
         break;
      case '}':
         x = stk.top();
         stk.pop();
         if (x=='(' || x=='[')
            return false;
         break;
      case ']':
         x = stk.top();
         stk.pop();
         if (x =='(' || x == '{')
            return false;
         break;
      }
   }
   return (stk.empty());
    }
};

File: Python/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        n=0
        for ch in s:
            n=n+1
            if ch=="(":
                if s[n]==")":
                    return True
                else: return False
            elif ch=="[":
                if s[n]=="]":
                    return True
                else: return False
            elif ch=="{":
                if s[n]=="}":
                    return True
                else: return False
            else: return False
